src/javascript/EditPanel/EditPanel.jsx

```diff
@@ -31,24 +31,36 @@ const EditPanelCmp = ({formik, title, notificationContext, client}) => {
             }
         };
 
-        if (formik.dirty) {
+        const registerListeners = () => {
             // Prevent close browser's tab when there is unsaved content
             window.addEventListener('beforeunload', handleBeforeUnloadEvent);
-        } else {
-            window.removeEventListener('beforeunload', handleBeforeUnloadEvent);
-        }
+            if (envProps.registerListeners) {
+                envProps.registerListeners(formik);
+            }
+        };
 
-        return () => {
+        const unregisterListeners = () => {
             window.removeEventListener('beforeunload', handleBeforeUnloadEvent);
+            if (envProps.unregisterListeners) {
+                envProps.unregisterListeners();
+            }
         };
+
+        if (formik.dirty) {
+            registerListeners();
```

Comment:
there is a double check to formik.dirty ? one here and one inside ContentEditor.redux.jsx (registerListeners)
May be we can remove the check from ContentEditor.redux.jsx (registerListeners) ?
and may be we do not need formik anymore inside ContentEditor.redux.jsx (registerListeners) ?

Reply:
Yes, the check was not useful. 
Thank you!